web_src/js/features/common-button.ts

```diff
@@ -155,12 +155,16 @@ function onShowModalClick(e) {
 }
 
 export function initGlobalButtons(): void {
+  initTargetButtons(document as ParentNode);
+}
+
+export function initTargetButtons(target: ParentNode): void {
   // There are many "cancel button" elements in modal dialogs, Fomantic UI expects they are button-like elements but never submit a form.
   // However, Gitea misuses the modal dialog and put the cancel buttons inside forms, so we must prevent the form submission.
   // There are a few cancel buttons in non-modal forms, and there are some dynamically created forms (eg: the "Edit Issue Content")
-  addDelegatedEventListener(document, 'click', 'form button.ui.cancel.button', (_ /* el */, e) => e.preventDefault());
+  queryElems(target, 'form button.ui.cancel.button', (el) => el.addEventListener('click', (e) => e.preventDefault()));
```

Comment:
NOOOOOOOOOOOO, do not do that.

It is not right. `addDelegatedEventListener` has its own clear purpose.

Reply:
It's been revised.